GolosChain/golos
# Add delete_reblog_operation to follow-plugin #751

#751 

Checklist:
1. Create 3 accounts and a comment to reblog:
```
./cli_wallet --server-rpc-endpoint=ws://127.0.0.1:8091
set_password qwer
unlock qwer
import_key 5JVFFWRLwz6JoP9kguuRFfytToGU6cLgBVTL9t6NB3D3BQLbUBS
create_account cyberfounder test "{}" "30.000 GOLOS" true
transfer_to_vesting cyberfounder test "30.000 GOLOS" true
post_comment  test test "" test hello world "{}" true

create_account cyberfounder testreb "{}" "30.000 GOLOS" true
transfer_to_vesting cyberfounder testreb "30.000 GOLOS" true
create_account cyberfounder testreb2 "{}" "30.000 GOLOS" true
transfer_to_vesting cyberfounder testreb2 "30.000 GOLOS" true
```

2. Follow on future rebloggers (need for feed_objects):
```
begin_builder_transaction
add_operation_to_builder_transaction 0 ["custom_json", {"required_posting_auths":["test"],"id": "follow", "json":"[\"follow\", {\"follower\":\"test\",\"following\":\"testreb\",\"what\":[\"blog\"]}]"}]
sign_builder_transaction 0 true
begin_builder_transaction
add_operation_to_builder_transaction 1 ["custom_json", {"required_posting_auths":["test"],"id": "follow", "json":"[\"follow\", {\"follower\":\"test\",\"following\":\"testreb2\",\"what\":[\"blog\"]}]"}]
sign_builder_transaction 1 true
```

3. Reblog
```
begin_builder_transaction
add_operation_to_builder_transaction 2 ["custom_json", {"required_posting_auths":["testreb"],"id": "follow", "json":"[\"reblog\", {\"account\":\"testreb\",\"author\":\"test\",\"permlink\":\"test\"}]"}]
sign_builder_transaction 2 true
begin_builder_transaction
add_operation_to_builder_transaction 3 ["custom_json", {"required_posting_auths":["testreb2"],"id": "follow", "json":"[\"reblog\", {\"account\":\"testreb2\",\"author\":\"test\",\"permlink\":\"test\"}]"}]
sign_builder_transaction 3 true
```

4. Check blog_index.
```
</script><script src="node_modules/golos-js/dist/golos.min.js"></script>
Hello world
<script>
golos.config.set('websocket', 'ws://127.0.0.1:8091');
var logger = function(err, result) { console.log(err, result); };
golos.api.getBlog("test", 0, 300, function (err, result) {
document.body.innerHTML += '<pre>' + JSON.stringify(result, null, 2) + "</pre>";
});
golos.api.getBlog("testreb", 0, 300, function (err, result) {
document.body.innerHTML += '<pre>' + JSON.stringify(result, null, 2) + "</pre>";
});
golos.api.getBlog("testreb2", 0, 300, function (err, result) {
document.body.innerHTML += '<pre>' + JSON.stringify(result, null, 2) + "</pre>";
});
</script>
```
Result: 3 blogs - by test, testreb, testreb2:

5. Check feed_index:
```
<script src="node_modules/golos-js/dist/golos.min.js"></script>
Hello world
<script>
golos.config.set('websocket', 'ws://127.0.0.1:8091');
var logger = function(err, result) { console.log(err, result); };
golos.api.getFeed("test", 0, 300, function (err, result) {
document.body.innerHTML = '<pre>' + JSON.stringify(result, null, 2) + "</pre>";
});
```
Result: feed_object with testreb and testreb2 in `reblog_by` field.

6. Delete first reblog:
```
begin_builder_transaction
add_operation_to_builder_transaction 4 ["custom_json", {"required_posting_auths":["testreb"],"id": "follow", "json":"[\"delete_reblog\", {\"account\":\"testreb\",\"author\":\"test\",\"permlink\":\"test\"}]"}]
sign_builder_transaction 4 true
```
Again, check blog_index and feed_index.
Result:
- testreb's blog is removed.
- testreb is removed from `reblog_by` field from feed_object.

7. Delete another reblog:
```
begin_builder_transaction
add_operation_to_builder_transaction 5 ["custom_json", {"required_posting_auths":["testreb2"],"id": "follow", "json":"[\"delete_reblog\", {\"account\":\"testreb2\",\"author\":\"test\",\"permlink\":\"test\"}]"}]
sign_builder_transaction 5 true
```
Again, check blog_index and feed_index.
Result:
- testreb2's blog is removed.
- feed_object is removed.
